src/libfetchers/git-utils.cc

```diff
@@ -644,17 +660,112 @@ struct GitInputAccessor : InputAccessor
     }
 };
 
-ref<InputAccessor> GitRepoImpl::getAccessor(const Hash & rev)
+struct GitExportIgnoreInputAccessor : FilteringInputAccessor {
+    ref<GitRepoImpl> repo;
+    std::optional<Hash> rev;
+
+    GitExportIgnoreInputAccessor(ref<GitRepoImpl> repo, ref<InputAccessor> next, std::optional<Hash> rev)
+        : FilteringInputAccessor(next, [&](const CanonPath & path) {
+            return RestrictedPathError(fmt("'%s' does not exist because it was fetched with exportIgnore enabled", path));
+        })
+        , repo(repo)
+        , rev(rev)
+    { }
+
+    bool gitAttrGet(const CanonPath & path, const char * attrName, const char * & valueOut) {
```

Comment:
```suggestion
    bool gitAttrGet(const CanonPath & path, const char * attrName, const char * & valueOut) 
    {
```

Instead of passing a `valueOut` by reference, isn't it cleaner to return an `std::optional<std::string>`? Then we don't have to worry about the ownership of that string.

Reply:
It'd be cleaner, but more expensive.